xl/migrations/database/to_bsddb.py

```diff
@@ -0,0 +1,63 @@
+# Copyright (C) 2017 Dustin Spicuzza
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2, or (at your option)
+# any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+#
+# The developers of the Exaile media player hereby grant permission
+# for non-GPL compatible GStreamer and Exaile plugins to be used and
+# distributed together with GStreamer and Exaile. This permission is
+# above and beyond the permissions granted by the GPL license by which
+# Exaile is covered. If you modify this code, you may extend this
+# exception to your version of the code, but you are not obligated to
+# do so. If you do not wish to do so, delete this exception statement
+# from your version.
+
+import logging
+import os
+import shelve
+
+from xl import common
+
+logger = logging.getLogger(__name__)
+
+def migrate(path):
+    bak_path = path + '.tmp'
+    
+    try:
+        old_shelf = shelve.open(path, 'r', protocol=common.PICKLE_PROTOCOL)
+    except Exception:
+        logger.warn("%s may be corrupt", path)
+        raise
+    
+    db = common.bsddb.hashopen(bak_path, 'c')
+    new_shelf = shelve.BsdDbShelf(db, protocol=common.PICKLE_PROTOCOL)
+    
+    for k,v in old_shelf.iteritems():
+        new_shelf[k] = v
+    
+    new_shelf.close()
+    old_shelf.close()
+    
+    try:
+        common.replace_file(bak_path, path)
```

Comment:
The fun thing here is that some dbm-style backends (I think dbm and dumbdbm) create multiple files (`music.db` is stored as `music.db.*`), and this would leave the old files behind.